src/pieces/counting.ts

```diff
@@ -0,0 +1,127 @@
+		if (msg.createdTimestamp <= countingDB.startDate) return;
+
+		const endMsg = `${msg.partial ? 'Someone' : msg.author.username} deleted a message and broke the count!`;
+
+		if (countingDB.count > 0) {
+			await endGame(endMsg, bot);
+		}
+	});
+
+	bot.on('messageCreate', async msg => {
+		if (msg.author.bot) return;
+
+		if (msg.channel.id === CHANNELS.COUNTING_CHANNEL) {
+			let gameOver = false;
+
+			if (msg.author.id === countingDB.lastSendId) {
+				gameOver = true;
+				msg.reply(`${msg.author.username}, you aren't allowed to count twice in a row!`);
+			}
+
+			countingDB.lastSendId = msg.author.id;
+
+			let endMsg = '';
+			if (msg.stickers.size > 0) {
+				gameOver = true;
+				endMsg = `${msg.author.username}, that's a nice sticker mate but we're supposed to be counting here.`;
+			} else if (msg.attachments.size > 0) {
+				gameOver = true;
+				endMsg = `${msg.author.username}, you can't send attachments here. This is a counting channel.`;
+			} else if (Number.isNaN(parseInt(msg.content))) {
+				gameOver = true;
+				endMsg = `Really ${msg.author.username}? ${msg.content} isn't a number!`;
+			} else if (parseInt(msg.content) > countingDB.count + 1) {
+				gameOver = true;
+				endMsg = `Come on ${msg.author.username}, ${msg.content} doesn't come after **${countingDB.count}**!`;
+			} else if (parseInt(msg.content) < countingDB.count) {
+				gameOver = true;
+				endMsg = `Believe it or not ${msg.author.username}, ${msg.content} is less than **${countingDB.count}**.`;
+			} else if (parseInt(msg.content) === countingDB.count) {
+				gameOver = true;
+				endMsg = `My dude that is literally the number we were on.`;
+			}
+
+			if (msg.content === (countingDB.count + 1).toString()) {
+				if (!countingDB.uniqueParticipants.includes(msg.author.id)) {
+					await bot.mongo.collection(DB.COUNTING).findOneAndUpdate({ piece: 'counting' }, { $push: { uniqueParticipants: msg.author.id } });
+				}
+
+				await bot.mongo.collection(DB.COUNTING).findOneAndUpdate({ piece: 'counting' }, { $inc: { count: +1 } });
+				await bot.mongo.collection(DB.COUNTING).findOneAndUpdate({ piece: 'counting' }, { $set: { lastSendId: msg.author.id } });
+				countingDB = await bot.mongo.collection(DB.COUNTING).findOne({ piece: 'counting' }); // update the counter data
+
+				if (countingDB.count === 69 || countingDB.count === 420) {
+					msg.reply('nice');
+				}
+			}
+
+			if (gameOver) {
+				await endGame(endMsg, bot);
+			}
+		}
+	});
+}
+
+async function endGame(message: string, bot: Client) {
+	const endDate = Date.now();
+	const resultEmbed = new MessageEmbed()
+		.setTitle(message)
+		.setDescription(`The count lasted for **${msToHMS(endDate - countingDB.startDate)}**. With **${countingDB.count}** counts and **${countingDB.uniqueParticipants.length}** participants ` +
+		`that's an average of one count every **${msToHMS((endDate - countingDB.startDate) / countingDB.count)}**`)
```

Comment:
This will fail if `countingDB.count` is zero. ~~this is an issue I have yet to fix in my implementation as well~~

Reply:
Good shout! It would still send but would say "The count lasted for 5.9 seconds. With 0 counts and 0 participants that's an average of one count every Infinity days, Infinity hours, NaN seconds".